mantis-common/src/main/java/io/mantisrx/runtime/descriptor/StageSchedulingInfo.java

```diff
@@ -29,12 +29,13 @@
 
 public class StageSchedulingInfo {
 
-    private int numberOfInstances;
-    private MachineDefinition machineDefinition;
-    private List<JobConstraints> hardConstraints;
-    private List<JobConstraints> softConstraints;
+    private final int numberOfInstances;
+    private final MachineDefinition machineDefinition;
+    private final List<JobConstraints> hardConstraints;
+    private final List<JobConstraints> softConstraints;
     private StageScalingPolicy scalingPolicy;
-    private boolean scalable;
+    private final boolean inheritInstanceCount;
```

Comment:
I am wondering if it makes sense semantically for inheritInstanceCount to be an attribute of the StageSchedulingInfo? Shouldn't this be more of an attribute of how the deployments are rolled out? Something like a DeploymentStrategy class and this being a member of that?

Disclaimer: I do not know a lot about this codebase, so many of these questions are based on semantics rather than specifics.

Reply:
<img width="953" alt="Screen Shot 2021-09-22 at 12 12 03 PM" src="https://user-images.githubusercontent.com/87735571/134406976-c0edc3b9-7528-46b2-b929-33e480666d3f.png">

Reply:
Note: this new DeploymentStrategy class is to provide better logical OO abstraction/isolation at the class level and it is not intended to extend the mantis master for large/complex deployment workflow.